tlr/tests-tlr/src/test/java/edu/kit/kastel/mcse/ardoco/tlr/tests/integration/evaluation/AbstractEvaluation.java

```diff
@@ -0,0 +1,102 @@
+/* Licensed under MIT 2025. */
+package edu.kit.kastel.mcse.ardoco.tlr.tests.integration.evaluation;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Locale;
+
+import org.eclipse.collections.api.factory.Lists;
+import org.eclipse.collections.api.list.MutableList;
+import org.junit.jupiter.api.Assertions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import edu.kit.kastel.mcse.ardoco.core.api.models.Metamodel;
+import edu.kit.kastel.mcse.ardoco.core.api.models.Model;
+import edu.kit.kastel.mcse.ardoco.core.api.output.ArDoCoResult;
+import edu.kit.kastel.mcse.ardoco.core.common.tuple.Pair;
+import edu.kit.kastel.mcse.ardoco.core.common.util.DataRepositoryHelper;
+import edu.kit.kastel.mcse.ardoco.core.tests.eval.ExpectedResults;
+import edu.kit.kastel.mcse.ardoco.metrics.result.SingleClassificationResult;
+
+abstract class AbstractEvaluation {
+    protected Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    protected void logExtendedResultsWithExpected(String name, SingleClassificationResult<?> results, ExpectedResults expectedResults) {
+        var infoString = String.format(Locale.ENGLISH, """
+
+                %s:
+                %s""", name, getExtendedResultStringWithExpected(results, expectedResults));
+        logger.info(infoString);
+    }
+
+    private String getExtendedResultStringWithExpected(SingleClassificationResult<?> result, ExpectedResults expectedResults) {
+        return "" +//
+                String.format(Locale.ENGLISH, """
+                        \tPrecision:%8.2f (min. expected: %.2f)
+                        \tRecall:%11.2f (min. expected: %.2f)
+                        \tF1:%15.2f (min. expected: %.2f)""", //
+                        result.getPrecision(), expectedResults.precision(), result.getRecall(), expectedResults.recall(), result.getF1(), expectedResults.f1()) //
+                + String.format(Locale.ENGLISH, """
+
+                        \tAccuracy:%9.2f (min. expected: %.2f)
+                        \tSpecificity:%6.2f (min. expected: %.2f)""",//
+                        result.getAccuracy(), expectedResults.accuracy(), result.getSpecificity(), expectedResults.specificity())//
+                + String.format(Locale.ENGLISH, """
+
+                        \tPhi Coef.:%8.2f (min. expected: %.2f)
+                        \tPhi/PhiMax:%7.2f (Phi Max: %.2f)
+                        %s""",//
```

Comment:
I am not a fan of having text blocks within the method call...

Reply:
Method just for formatting ...